PigLatin/PigLatin.cs

```diff
@@ -2,20 +2,57 @@
 
 namespace PigLatin
 {
-    class Program
+  class Program
+  {
+    public static void Main()
     {
-        public static void Main()
-        {
-            // your code goes here
+      Console.WriteLine("Enter a word to be translated to Pig Latin: ");
+      string word = Console.ReadLine();
+      Console.WriteLine(Translate(word));
+      Console.WriteLine("Now, enter a sentence to be translated to Pig Latin: ");
+      string sentence = Console.ReadLine();
+      string[] separateWords = sentence.Split(' ');
+      for (int i = 0; i < separateWords.Length; i++)
+      {
+        separateWords[i] = Translate(separateWords[i].ToLower());
+      }
+      Console.WriteLine(String.Join(' ', separateWords));
+    }
 
-            // leave this command at the end so your program does not close automatically
-            Console.ReadLine();
+    static string Translate(string word)
+    {
+
+      int halfWordLength = word.Length / 2;
+      Convert.ToInt32(halfWordLength);
+      string firstHalf = word.Substring(0, halfWordLength);
+      string lastHalf = word.Substring(halfWordLength);
+      int firstLetterIndex = word.IndexOfAny(new char[] { 'a', 'e', 'i', 'o', 'u' });
+      if (firstLetterIndex == 0)
+      {
+        if (word[word.Length - 1] == 'a' || word[word.Length - 1] == 'e' || word[word.Length - 1] == 'i' || word[word.Length - 1] == 'o' || word[word.Length - 1] == 'u')
+        {
+          return (word + "yay");
         }
-        
-        public static string TranslateWord(string word)
+        else
         {
-            // your code goes here
-            return word;
+          return (word + "ay");
         }
+      }
+      else if (!(word.Contains("a") || word.Contains("e") || word.Contains("i") || word.Contains("o") || word.Contains("u")))
+      {
+        int findY = word.IndexOf("y");
+        string beforeY = word.Substring(0, findY);
```

Comment:
This will also throw an exception for words that have no vowels and no 'y'. 
Try "tsk".